src/main/scala/io/epiphanous/flinkrunner/model/sink/JdbcSinkConfig.scala

```diff
@@ -614,8 +621,9 @@ case class JdbcSinkConfig[ADT <: FlinkEvent](
               case ts: Instant       => Timestamp.from(ts)
               case Some(ts: Instant) => Timestamp.from(ts)
               case Some(x)           => x
+              case Some(m: Map[String, Any]) => encoder.encode(m)
```

Comment:
So, the previous `case` will swallow this one, meaning line 624 won't ever get executed. To solve this, I'd rewrite the match expressions as a recursive partial function where we have cases handling the options first, i.e., a `case None => null` and `case Some(x) => func(x)` where `func` is referring to the name of the recursive partial function. Then put all the non-option cases below. Hopefully we can make the partial function tail recursive.